test/unit/dynamic-variables.test.js

```diff
@@ -86,5 +86,14 @@ describe('Dynamic variable', function () {
             expect(directoryPath).to.not.be.undefined;
             expect(directoryPath).to.not.be.null;
         });
+
+        it('$isoTimestamp returns a timestamp in ISO format', function () {
+            var isoTimestamp = dynamicVariables.$isoTimestamp.generator();
+
+            expect(isoTimestamp).to.be.a('string');
+            expect(isoTimestamp).to
+                // eslint-disable-next-line max-len
+                .match(/^([+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24:?00)([.,]\d+(?!:))?)?(\17[0-5]\d([.,]\d+)?)?([zZ]|([+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/);
```

Comment:
Is there a way to simplify this regex? This is unreadable and we have no idea what/how it's testing.

Reply:
This covers many time-related corner cases. 
Let's update it with a simpler one. I will also add comments to it.